### ISSUE TITLE
Add --cloud flag to panel serve (I.e. provide deployment solution)

### ISSUE THREAD

**Issue by MarcSkovMadsen** [holoviz/panel#3641]
## Context

Trying to make it easy for Panel users to deploy and share their work. Inspired by the lightning.ai. To run an app locally you just write `lighting run app app.py`. If you want to deploy you run `lightning run app app.py --cloud`. 

If you have not logged in or registered, then a url will show up to enable you to login/ register with your google or github account. That is sooooooo easy.

See https://github.com/Lightning-AI/lightning/issues/13335

![image](https://user-images.githubusercontent.com/42288570/174946386-60007248-86b7-4594-9871-cec0892d36e9.png)

## Feature Request

Provide something similar for Panel.

## Additional Context

I know building a cloud offering takes time and money. But maybe there could be ways to integrate with some cloud providers take make the experience just as easy.

Maybe the scope could be pyscript only apps.

Getting the time to "production" down is so important for Python and data science in general.

**Comment by MarcSkovMadsen** [comment]
FYI @Hoxbro . The API of lightning.ai apps powers FastAPI. There might be some inspiration for you in the api and the way they use FastAPI.

**Comment by MarcSkovMadsen** [comment]
With Anacondas acquisition of PythonAnywhere I guess I could dream of a command `panel serve app.py --pythonanywhere`.

**Comment by jbednar** [comment]
Yes, you can dream that! But dreams can take time to become reality. :-)
